src/fb.d.ts

```diff
@@ -0,0 +1,67 @@
+interface FBOptions {
+    accessToken: string;
+    appId: string;
+    appSecret: string;
+    version?: string;
+    proxy?: string;
+    timeout?: number;
+    scope?: string;
+    redirectUri?: string;
+    Promise?: Promise<any>;
+}
+
+interface FBAPIInterface {
+    path: string;
+    method?: string;
+    params?: object;
+    callback?: Function;
+}
+
+interface FBKeyOrOptions {
+    key: string;
+    options: FBOptions;
+}
+
+interface SignedRequestResponse {
+    oauthToken: string;
+    userId: string;
+    userCountry: string;
+}
+
+interface UsageInterface {
+    callCount: number;
+    totalTime: number;
+    totalCPUTime: number;
+}
+
+export class Facebook {
+    constructor(options?: FBOptions);
+
+    api(arguments: FBAPIInterface): Promise<any>;
```

Comment:
api and napi args are not a single arguments object, they're separate ordered arguments.

You'll need overloading here as well to differentiate between `path, cb` and `path, params, cb`, etc and also to differentiate between the callback form and the form without the callback that returns a promise.